File: C_Projects/C_Euler/src/5.c
#include "C_Euler.h"

int64 Euler_5() {
	int count = 0;
	while (1) {
		count += 2 * 2 * 2 * 2 * 3 * 3 * 5 * 7 * 11 * 13 * 17 * 19;
		int num = 2;
		while (num <= 20) {
			if (count % num != 0) {
				break;
			}
			++num;
		}
		if (num == 21) {
			return (int64) count;
		}
	}
}

File: Py_Projects/Euler/5.py
def Euler_5(num=20):
	'''Find the smallest integer evenly divisible by all numbers up to the input
	'''
	factor_list = []
	for x in range(1,num+1):
		for i in factor_list:
			if x % i == 0:
				x //= i
		if x != 1:
			factor_list.append(x)
	total = 1
	for x in factor_list:
		total *= x
	return total

if __name__ == '__main__':
	print(Euler_5())
